File: Difficulty: Basic/Queue Push & Pop/queue-push-pop.cpp
//{ Driver Code Starts
// Initial Template for C++
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

queue<int> _push(vector<int> &arr);

void _pop(queue<int> &s);


// } Driver Code Ends

// User function Template for C++

queue<int> _push(vector<int> &arr) {
    // Create a queue and insert all elements from the vector into it
    queue<int> q;
    for(int i = 0; i < arr.size(); i++) {
        q.push(arr[i]);
    }
    return q;
}

void _pop(queue<int> &q) {
    // Print the front element and remove it from the queue until it becomes empty
    while(!q.empty()) {
        cout << q.front() << " ";
        q.pop();
    }
}


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        string str;
        getline(cin, str);
        stringstream ss(str);
        vector<int> arr;
        int num;
        while (ss >> num) {
            arr.push_back(num);
        }
        queue<int> mys = _push(arr);
        _pop(mys);
        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Basic/Queue Push & Pop/queue-push-pop.js
//{ Driver Code Starts
//Initial Template for javascript

"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin;
});

process.stdin.on("end", (_) => {
  inputString = inputString
    .trim()
    .split("\n")
    .map((string) => {
      return string.trim();
    });

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/* Function to print an array */
function printArray(arr, size) {
  let i;
  let s = "";
  for (i = 0; i < size; i++) {
    if (arr[i] == -0) arr[i] = 0;
    s += arr[i] + " ";
  }
  console.log(s);
}

function main() {
  let t = parseInt(readLine());
  let i = 0;
  for (; i < t; i++) {
    let n = parseInt(readLine());
    let s = "";
    let arr = readLine()
      .split(" ")
      .map((x) => parseInt(x));
    let ob = new Solution();
    let queue = ob._push(arr, n);
    ob._pop(queue);
  
console.log("~");
}
}

// } Driver Code Ends

// User function Template for JavaScript

class Solution {
  /**
   * @param {number[]} arr
   * @param {number} n
   * @return {number[]}
   */
  _push(arr, n) {
    // Initialize an empty queue (array)
    let queue = [];
    
    // Insert all elements of arr into the queue
    for (let i = 0; i < n; i++) {
      queue.push(arr[i]);
    }

    // Return the filled queue
    return queue;
  }

  /**
   * @param {number[]} queue
   */
  _pop(queue) {
    // Initialize an array to store dequeued elements
    let dequeuedElements = [];

    // Dequeue elements one by one and store them
    while (queue.length > 0) {
      dequeuedElements.push(queue.shift());
    }

    // Print the dequeued elements space-separated
    console.log(dequeuedElements.join(" "));
  }
}
